core/src/main/java/org/mule/api/debug/ObjectFieldDebugInfo.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.api.debug;
+
+import static org.mule.util.Preconditions.checkArgument;
+
+import java.util.List;
+
+/**
+ * Provides debug information for an object's field that contains a value
+ * composed of many simple values.
```

Comment:
Not 100% sure what this means, does this means this is used to provide values for all List,Collection,Array etc?  What about maps?